src/instruction_handlers/storage.rs

```diff
@@ -16,6 +17,14 @@ fn sstore<T: Tracer, W: World<T>>(
     instruction_boilerplate::<opcodes::StorageWrite, _, _>(vm, world, tracer, |vm, args, world| {
         let key = Register1::get(args, &mut vm.state);
         let value = Register2::get(args, &mut vm.state);
+        if !vm
+            .world_diff
+            .written_storage_slots
+            .contains(&(vm.state.current_frame.address, key))
```

Comment:
These checks add significant overhead and are pretty ugly. Would it be possible to instead look at the written storage slots at the end? All checks seem to just trigger on the first time something is added, except the one check that checks both reads and writes.